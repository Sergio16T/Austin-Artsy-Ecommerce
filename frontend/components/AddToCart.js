import React, { Component } from 'react';
import { Mutation } from 'react-apollo'; 
import gql from 'graphql-tag'; 
import { CURRENT_USER_QUERY } from './User'; 

const ADD_TO_CART_MUTATION = gql`
    mutation ADD_TO_CART_MUTATION($id: ID! ) {
        addToCart(id: $id) {
            id
            quantity
        }
    }
`; 

class AddToCart extends Component {
    render() {
        return (
            <Mutation 
            mutation={ADD_TO_CART_MUTATION} 
            variables={{id: this.props.id}}
            refetchQueries={[{query: CURRENT_USER_QUERY }]}
            >
                {(addToCart, {error, loading}) => 
                    <button type="button" onClick={async () => {
                        const res = await addToCart().catch(err => {
                            alert(err.message); 
                        }); 
                        console.log(res);
                    }}>Add to Cart</button>
                }
            </Mutation>
        );
    }
}

export default AddToCart;