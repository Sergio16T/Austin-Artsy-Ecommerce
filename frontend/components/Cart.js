import React, { Component } from 'react';
import styled from 'styled-components';

const CartWrapper = styled.div`
    position: fixed; 
    height: 100%; 
    width: 400px; 
    transform: ${props => props.isOpen ? "translateX(0)" : "translateX(100%)" }; 
    background: ${props => props.theme.offWhite}; 
    right: 0; 
    top: 0; 
    z-index: 5; 
    border-left: 2px solid ${props => props.theme.purple}; 
    min-width: 500px; 
    transition: all 0.3s;
    overflow: hidden; 
`; 

class Cart extends Component {
    render() {
        return (
            <CartWrapper isOpen={this.props.isOpen}>
                <h2>Cart</h2>
            </CartWrapper>
        );
    }
}

export default Cart;