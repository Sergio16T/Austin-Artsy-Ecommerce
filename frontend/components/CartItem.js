import React from 'react';
import { useMutation } from 'react-apollo'; 
import gql from 'graphql-tag'; 
import styled from 'styled-components';
import RemoveFromCart from './RemoveFromCart'; 
import formatMoney from "../lib/formatMoney";
import { CURRENT_USER_QUERY } from './User';

const UPDATE_CART_ITEM_MUTATION = gql`
    mutation UPDATE_CART_ITEM_MUTATION($updateType: String!, $id: ID!) {
        updateCartItem(updateType: $updateType, id: $id) {
            id
            quantity
        }
    }
`; 

const StyledCartItem = styled.div`
display: grid; 
grid-template-columns: 1fr 2fr .5fr; 
padding: 2rem 0; 
.cartItemImage {
    width: 120px; 
}
@media (max-width: 600px) {
    padding: 1rem 0; 
    .cartItemImage {
        width: 80px; 
    }
}
`; 

const CartItemDescription = styled.div`
    padding-left: 2rem;
.cartitem_title{
    margin-top: 0; 
    font-size: 1.2rem;
}
.priceandquantity {
    display: flex; 
    flex-direction: row; 
}
`; 
const QuantityDiv = styled.div`
    display: flex;
    flex-direction: row; 
    font-size: 1.6rem; 
    padding-left: 2rem; 
    align-items: center; 
.cartItem_Quantity, .minus, .plus {
    padding:  0 .75rem; 
    margin: 0; 
}
.minus, .plus {
    outline: none; 
    border: none; 
    background: transparent; 
    font-size: 1.6rem; 
    cursor: pointer; 
}
.cartitem_price {
    font-size: 1.2rem; 
}
`; 
const CartItem = (props) => {
    const [updateCartItem, { data }] = useMutation(UPDATE_CART_ITEM_MUTATION); 
	const { item } = props.cartItem; 
	return (
		<StyledCartItem>
			<img className="cartItemImage" src={item.image}/> 
			<CartItemDescription>
				<p className="cartitem_title">{item.title}</p>
				<div className="priceandquantity">
					<p className="cartitem_price">{formatMoney(item.price)}</p>
					<QuantityDiv>
						<button className="minus" type="button" onClick={() => {
                            updateCartItem({variables: { updateType: "subtract", id: props.cartItem.id}}); 
                        }} disabled={props.cartItem.quantity === 1}>-</button>
						<p className="cartItem_Quantity">{props.cartItem.quantity}</p>
						<button className="plus" onClick={() => {
                            updateCartItem({variables: { updateType: "add", id: props.cartItem.id}}); 
                        }}>+</button>
					</QuantityDiv>
				</div>
			</CartItemDescription>
			
			<RemoveFromCart id={props.cartItem.id}/>
		</StyledCartItem>
	)
}

export default CartItem; 