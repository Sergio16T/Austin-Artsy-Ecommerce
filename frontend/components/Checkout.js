import React, { Component } from 'react';
import StripeCheckout from 'react-stripe-checkout'; 
import { Mutation } from 'react-apollo'; 
import Router from 'next/router'; 
import Nprogress from 'nprogress'; 
import PropTypes from 'prop-types'; 
import gql from 'graphql-tag'; 
import calculateTotal from '../lib/calculateTotal'; 
import { CURRENT_USER_QUERY } from './User'; 

const CREATE_ORDER_MUTATION = gql`
    mutation CREATE_ORDER_MUTATION($token: String!) {
        createOrder(token: $token) {
            id
						total 
						charge
						items {
							id
							title
						}
        }
    }
`; 

function totalItems(cart) {
    return cart.reduce((startValue, element) => {
        return startValue + element.quantity; 
    }, 0); 
}

class Checkout extends Component {
    onToken = async (res, createOrder) => {
				console.log(res.id); 
				this.props.toggleCart(false);
				const order = await createOrder({
					variables: { token: res.id }
				}).catch(err => {
					console.log(err.message);  
				}); 
				console.log(order); 
				Router.push({
					pathname: "/order", 
					query: {id: order.data.createOrder.id}
				});
			
    }
    render() {
        return (
						<Mutation 
						mutation={CREATE_ORDER_MUTATION}
						refetchQueries={[{ query: CURRENT_USER_QUERY }]}
						>
                {(createOrder, {error, loading}) => 
							<StripeCheckout
							amount={this.props.totalAmount}
							name="Austin's List"
							description={`Order of ${totalItems(this.props.cart)} items`}
							image={this.props.cart.length && this.props.cart[0].item ? this.props.cart[0].item.image : null}
							stripeKey="pk_test_lUHGmLxt0KGqdsIN20tuXiVw00Q2OwgjiR"
							currency="USD"
							email={this.props.email}
							token={(res) => this.onToken(res, createOrder)}
							>{this.props.children}</StripeCheckout>
            }
            </Mutation>
        );
    }
}

export default Checkout;