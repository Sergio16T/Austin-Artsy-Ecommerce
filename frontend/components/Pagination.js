import React, { Component } from 'react';
import PaginationStyles from './styles/PaginationStyles'; 
import { Query } from 'react-apollo'; 
import Head from "next/head"; 
import Link from "next/link"; 
import styled from 'styled-components'; 
import gql from 'graphql-tag'; 

const perPage = 4; 

const PAGINATION_QUERY = gql`
    query PAGINATION_QUERY{
        itemsConnection {
            aggregate {
                count
            }
        }
    }
`;


const PaginationContainer = styled.div`
    padding-top: 4rem; 
    width: 100%; 
    display: flex; 
    justify-content: center; 
`;
const Placeholder = styled.div`
    height: 45px;
    margin-top: 24px; 
    margin-bottom: 24px; 
`; 
const Pagination = props => {
    return (
        <PaginationContainer>
            <Query query={PAGINATION_QUERY}>
                {({data, error, loading}) => {
                    if(loading) return <Placeholder/>; 
                    if(error) return <p>{error.message}</p>
                    const count = data.itemsConnection.aggregate.count; 
                    const pages = Math.ceil(count / perPage); 
                    return (
                    <PaginationStyles data-testid="pagination">
                        <Head>
                            <title>Austin's List page {props.page} of {pages}</title>
                        </Head>
                        <Link                      
                        href={{
                            pathname: "/items",
                            query: {page: props.page -1}
                        }}
                        >
                        <a aria-disabled={props.page <= 1} id="prev"> &#8592; Prev</a>
                        </Link>
                        <p data-test="pagecount">Page {props.page} of {pages}</p>
                        <Link                      
                        href={{
                            pathname:"/items",
                            query: {page: props.page + 1}
                        }}

                        >
                        <a aria-disabled={props.page >= pages} id="next">Next &#8594;</a>
                        </Link>
                    </PaginationStyles>
                    )}}
            </Query> 
        </PaginationContainer>
    );
};


export default Pagination;
export { PAGINATION_QUERY }; 