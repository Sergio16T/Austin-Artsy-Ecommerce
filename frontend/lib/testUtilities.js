import casual from 'casual'; 
//.seed() is to provide ability to use a specific seed in order to get a repeatable random sequence for our testing purposes
casual.seed(111); 

const mockItem = () => ({
    // __typename: 'Item',
    id: '123',
    price: 5000,
    user: null,
    image: 'dog.jpg',
    title: 'dogs are the best',
    description: 'dogs description',
    largeImage: 'dog-large.jpg',
}); 

const mockUserWithItemsInCart = () =>  ({
    __typename: 'User', 
    id: "123", 
    name: casual.name, 
    email: casual.email, 
    permissions: ["ADMIN"], 
    cart: [{
        id: 'abc', 
        quantity: 2, 
        item: {
            id: 'dgh', 
            price: 15000, 
            image: 'www.dog.image.com',
            title: "puppy"
        }
        },
    ]
}); 
const mockUser = () => ({
    __typename: 'User', 
    id: "123", 
    name: casual.name, 
    email: casual.email, 
    permissions: ['ADMIN'], 
    cart: [],
    orders: []
}); 
const mockOrderItem = () => ({
    __typename: 'OrderItem',
    id: casual.uuid,
    image: `${casual.word}.jpg`,
    title: casual.words(),
    price: 4234,
    quantity: 1,
    description: casual.words(),
  });
  
  const mockOrder = () => ({
    __typename: 'Order',
    id: 'ord123',
    charge: 'ch_123',
    total: 40000,
    items: [mockOrderItem(), mockOrderItem()],
    createdAt: '2018-04 - 06T19: 24: 16.000Z',
    user: mockUser(),
  });
  
  const mockCartItem = overrides => ({
    __typename: 'CartItem',
    id: 'omg123',
    quantity: 3,
    item: mockItem(),
    user: mockUser(),
    ...overrides,
  });
//mock LocalStorage 
class MockLocalStorage {
    constructor(){
        this.store={}; 
    }

    clear() {
        this.store = {}; 
    }
    getItem(key) {
        return this.store[key] || null; 
    }
    setItem(key, value) {
        this.store[key] = value.toString(); 
    }
    removeItem(key) {
        delete this.store[key]; 
    }
}

export { 
    mockItem, 
    MockLocalStorage, 
    mockUser, 
    mockCartItem, 
    mockOrderItem, 
    mockOrder }; 