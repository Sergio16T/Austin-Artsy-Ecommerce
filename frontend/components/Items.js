import React, { Component } from 'react';
import { Query } from 'react-apollo'; 
import gql from 'graphql-tag';
import Item from './Item';
import { ItemStyles, StyledItemsWithPagination, StyledItemContainer } from './styles/ItemStyles'; 
import Pagination from './Pagination'; 

const ALL_ITEMS_QUERY = gql`
    query ALL_ITEMS_QUERY($skip: Int = 0, $first: Int = 4) {
        items(skip: $skip, first: $first, orderBy: createdAt_ASC) {
            id
            title
            price
            description
            image 
            largeImage
        }
    }
`; 


class Items extends Component {
    render() {
        return (
            <ItemStyles>
                <Query query={ALL_ITEMS_QUERY}
                variables={{skip: this.props.page * 4 - 4}}
                >
                    {({ data, error, loading }) => { 
                        if (loading) return null; 
                        if (error) return null; 
                        // console.log('payload', data);
                        return (
                            <StyledItemsWithPagination>
                                <Pagination page={this.props.page}/>
                                        <StyledItemContainer>    
                                            {data.items.map(item => (
                                                <Item 
                                                key={item.id} 
                                                item={item} 
                                                page={this.props.page}
                                                toggleCart={this.props.toggleCart}
                                                />           
                                            ))}
                                        </StyledItemContainer>  
                                <Pagination page={this.props.page}/>
                            </StyledItemsWithPagination>
                        )
                    }}
                </Query>
            </ItemStyles>
        );
    }
}

export default Items;
export { ALL_ITEMS_QUERY }
