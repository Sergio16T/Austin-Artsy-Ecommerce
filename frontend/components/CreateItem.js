import React, { Component } from 'react';
import { Query, Mutation } from 'react-apollo'; 
import Router from 'next/router'; 
import gql from'graphql-tag'; 
import { StyledFormWrapper, StyledForm} from './styles/FormStyles'; 
import Spinner from './Spinner'; 
import { ALL_ITEMS_QUERY } from './Items'; 
import { PAGINATION_QUERY } from './Pagination'; 

const CREATE_ITEM_MUTATION = gql`
    mutation CREATE_ITEM_MUTATION(
        $title: String! 
        $description: String!
        $price: Int! 
        $image: String!
        $largeImage: String!
        ) {
        createItem(
            title: $title, 
            description: $description, 
            price: $price
            image: $image
            largeImage: $largeImage 
            ) {
            id
        }
    }
`; 
class CreateItem extends Component {
    state = {
        title: '', 
        price: 0, 
        description: '',
        image: '',
        largeImage: '',
        errorMessage: '',
        uploadError: ''
    }
    uploadFile = async (e) => {
        console.log('uploading file'); 
        this.setState({ spinner: true }); 
        const files = e.target.files; 
        const data = new FormData(); 
        data.append('file', files[0]); 
        data.append('upload_preset', 'AustinArts'); 

        const res = await fetch('https://api.cloudinary.com/v1_1/dddnhychw/image/upload', {
            method: 'POST', 
            body: data  
        }); 
        const file = await res.json(); 
        console.log(file); 

        try {
            this.setState({
                image: file.secure_url, 
                largeImage: file.eager[0].secure_url,
                spinner: false
            }); 
        }
        catch(err) {
            this.setState({uploadError: err.message}); 
        }
    }
    handleInput = (e) => {
        let regEx = /^\d+(\.\d{0,2})?$/; 
        const { value, type, name } = e.target; 
        let val = type === 'number' && value.length ? parseFloat(value) : value;  
        // console.log(regEx.test(val)); 
        try {
            if ((typeof val === 'number') && !regEx.test(val)) {
                throw new Error('Only two decimal points please'); 
            }
        }
        catch(err) {
            console.log(err.message);
            this.setState({
                errorMessage: err.message
            }); 
        }
        if ((typeof val === 'number') && regEx.test(val)) {
            this.setState({
                errorMessage: ''
            }); 
        }
        this.setState({
            [name]: val
        }); 
    }
    render() {
        let regEx2= /^\d+(\.\d{0,1})$/; 
        const { errorMessage } = this.state; 
        const setPrice = regEx2.test(this.state.price) ? this.state.price.toFixed(2) : this.state.price; 
        return (
            <StyledFormWrapper>
                <div className="formContainer">
                    <Query query={PAGINATION_QUERY}>
                        {({data, error, loading}) => {
                            if(loading) return null; 
                            const count = data.itemsConnection.aggregate.count; 
                            // console.log(count); 
                            let lastOnPage  = count/4 === 0; 
                            let page = Math.ceil((count+1)/4) 
                            // console.log(page); 
                            let skip = page * 4 - 4;
                            // console.log(skip); 
                            return (
                        <Mutation 
                        mutation={CREATE_ITEM_MUTATION} 
                        variables={{...this.state, price: setPrice * 100}}
                        refetchQueries={[
                            { query: PAGINATION_QUERY }, 
                            { query: ALL_ITEMS_QUERY, variables: {skip: skip} }, 
                            { query: ALL_ITEMS_QUERY, variables: {skip: (page-1) * 4 - 4 }}
                        ]}
                        >
                            {(createItem, {loading, error}) => (
                            <StyledForm onSubmit={ async (e)=> {
                                e.preventDefault(); 
                                this.setState({ spinner: true });
                                let res = await createItem().catch(err => {
                                    if(err) {
                                        this.setState({ 
                                            spinner: false, 
                                            error: true
                                         }); 
                                    } else {
                                        this.setState({
                                            error: false
                                        }); 
                                    }  
                                }); 
                                this.setState({ spinner: false }); 
                                console.log(res); 
                                if(!this.state.error) {
                                    Router.push({
                                        pathname: "/items", 
                                        query: {page: page}
                                        // query: {id: res.data.createItem.id}
                                    }); 
                                }
                            }} 
                            errorMessage = {errorMessage}
                            >
                                <Spinner spinner={this.state.spinner}/>
                                {error && <p className="errorMessage">{error.message.replace("GraphQL error:", "")}</p>}
                                <fieldset disabled={loading} aria-busy={loading}>
                                    <label htmlFor="file">
                                        Image
                                        <input
                                        type ="file" 
                                        id ="file" 
                                        name="file" 
                                        placeholder="upload an image" 
                                        onChange={this.uploadFile}
                                        />
                                    </label>
                                    {this.state.largeImage && <img width="250" src={this.state.image} alt={this.state.description}></img>}
                                    <label htmlFor="title">
                                        Title
                                        <input
                                        type ="text" 
                                        id ="title" 
                                        name="title" 
                                        placeholder="title" 
                                        value={this.state.title}
                                        onChange={this.handleInput}
                                        required  
                                        />
                                    </label>
                                    <label htmlFor="price" id="priceLabel">
                                        {errorMessage ? errorMessage : 'Price'}
                                        <input
                                        type ="number" 
                                        id ="price" 
                                        name="price" 
                                        placeholder="price" 
                                        value={this.state.price}
                                        step=".01"
                                        onChange={this.handleInput}
                                        required  
                                        />
                                    </label>
                                    <label htmlFor="description">
                                        Description
                                        <textarea
                                        id ="description" 
                                        name="description" 
                                        placeholder="Enter a Description" 
                                        value={this.state.description}
                                        onChange={this.handleInput}
                                        required  
                                        />
                                    </label>
                                    <button type="submit" disabled={errorMessage ? true : false} aria-disabled={errorMessage ? true : false}> Submit </button>
                                </fieldset>
                            </StyledForm>
                        )}
                        </Mutation> 
                    )}}
                </Query>
                </div>
            </StyledFormWrapper>
        );
    }
}

export default CreateItem;