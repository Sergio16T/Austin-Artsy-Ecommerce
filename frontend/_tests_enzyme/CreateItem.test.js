import { mount } from 'enzyme'; 
import wait from 'waait'; 
import { shallowToJson } from 'enzyme-to-json'; 
import { MockedProvider } from '@apollo/react-testing'; 
import CreateItem, { CREATE_ITEM_MUTATION } from '../components/CreateItem'; 
import { PAGINATION_QUERY } from '../components/Pagination'; 
import { ALL_ITEMS_QUERY } from '../components/Items'; 
import { mockUser } from '../lib/testUtilities'; 
import Router from 'next/router'; 

const mockItem2 = {
    __typename: 'Item',
    id: '123',
    user: mockUser(),
    title: 'Spring Fling Trainers', 
    description: 'High Top Trainers', 
    price: 15000, 
    image: 'www.SFT.cloudinary.com', 
    largeImage: 'www.SFTLG.cloudinary.com'
}; 
//mock the global fetch API 
global.fetch = jest.fn().mockResolvedValue({
    json: () => ({
        secure_url: 'www.SFT.cloudinary.com',
        eager: [{ secure_url: 'www.SFTLG.cloudinary.com' }]
    }) 
});

const mocks =[{
        request: { query: PAGINATION_QUERY }, 
        result: { 
            data: {
                itemsConnection: {
                    __typename: 'aggregate',
                    aggregate: {
                        __typename: 'count', 
                        count: 10
                    }
                },
            },
        },
    },
    { request: {
        query: CREATE_ITEM_MUTATION, 
        variables: { 
            title: mockItem2.title, 
            description: mockItem2.description, 
            price: mockItem2.price,  
            image: 'www.SFT.cloudinary.com',
            largeImage: 'www.SFTLG.cloudinary.com'
        }
    },
    result: {
        data: {
            createItem: {
                ...mockItem2, 
                __typename: 'Item', 
            },
        },
        // refetch: () => console.log('refetch mocked') 
    },  
}
]
  
describe('<CreateItem/>', () => {
    it('renders and matches snapshot', async() => {
        const wrapper = mount(
            <MockedProvider mocks={mocks}>
                <CreateItem />
            </MockedProvider>
        ); 
        await wait(); 
        wrapper.update(); 
        // console.log(wrapper.debug()); 
        const form = wrapper.find('form[data-test="form"]'); 
        // console.log(form.debug()); 
        expect(shallowToJson(form)).toMatchSnapshot(); 
    }); 
    it('uploads a file ', async () => {
        const wrapper = mount(
            <MockedProvider mocks={mocks}>
                <CreateItem />
            </MockedProvider>
        ); 
        await wait(); 
        wrapper.update(); 
        const input = wrapper.find('input[type="file"]');
        input.simulate('change', {target: { files: ['SpringFling.jpg'] }}); 
        await wait(); 
        const component = wrapper.find('CreateItem').instance(); 
        // console.log(component); 
        expect(component.state.image).toEqual('www.SFT.cloudinary.com'); 
        expect(component.state.largeImage).toEqual('www.SFTLG.cloudinary.com'); 
        expect(global.fetch).toHaveBeenCalled(); 
        //.toHaveBeenCalledWith also an option to test here 
        // wrapper.update(); 
        global.fetch.mockReset(); 
    }); 
    it('updates state', async () => {
        const wrapper = mount(
            <MockedProvider mocks={mocks}>
                <CreateItem />
            </MockedProvider>
        ); 
        await wait(); 
        wrapper.update(); 
        wrapper.find('#title').simulate('change', { target: { name: 'title', value: mockItem2.title}}); 
        wrapper.find('#description').simulate('change', { target: { name: 'description', value: mockItem2.description }}); 
        wrapper.find('#price').simulate('change', { target: { name: 'price', value: 150 }}); 

        expect(wrapper.find('CreateItem').instance().state).toMatchObject({
            title: "Spring Fling Trainers", 
            price: 150, 
            description: "High Top Trainers"
        }); 

    }); 
    // it('creates an item when submited', async () => {
    //     const wrapper = mount(
    //         <MockedProvider mocks={mocks} >
    //             <CreateItem />
    //         </MockedProvider>
    //     ); 
    //     global.fetch = jest.fn().mockResolvedValue({
    //         json: () => ({
    //             secure_url: 'www.SFT.cloudinary.com',
    //             eager: [{ secure_url: 'www.SFTLG.cloudinary.com' }]
    //         }) 
    //     });
    //     await wait(); 
    //     wrapper.update(); 
    //     wrapper.find('#title').simulate('change', { target: { name: 'title', value: mockItem2.title }}); 
    //     wrapper.find('#description').simulate('change', { target: { name: 'description', value: mockItem2.description }}); 
    //     wrapper.find('#price').simulate('change', { target: { name: 'price', value: 150 }}); 


    //     const input = wrapper.find('input[type="file"]');
    //     input.simulate('change', {target: { files: ['SpringFling.jpg'] }}); 
    //     await wait(); 
    //     wrapper.update(); 
    //     const component = wrapper.find('CreateItem').instance(); 
    //     // console.log(component); 
    //     expect(component.state.image).toEqual('www.SFT.cloudinary.com'); 
    //     expect(component.state.largeImage).toEqual('www.SFTLG.cloudinary.com'); 
    //     expect(global.fetch).toHaveBeenCalled(); 

    //     expect(wrapper.find('CreateItem').instance().state).toMatchObject({
    //         title: "Spring Fling Trainers", 
    //         price: 150, 
    //         description: "High Top Trainers",
    //         image: 'www.SFT.cloudinary.com',
    //         largeImage: "www.SFTLG.cloudinary.com"
    //     }); 
    //     console.log(wrapper.find('CreateItem').instance().state); 
    //     console.log(wrapper.debug()); 
    //     // error is being caught in front end.. not sure.. No more mocked responses for the query:
    //     Router.router = { push: jest.fn() }; 
    //     // const submit = wrapper.find('[type="submit"]'); 
    //     // console.log('FORMMM', submit.debug()); 
    //     // submit.simulate('submit');
    //     wrapper.find('form').simulate('submit');
    //     await wait(50); 
    //     // wrapper.update(); 
    //     expect(Router.router.push).toHaveBeenCalled(); 
    // })
})