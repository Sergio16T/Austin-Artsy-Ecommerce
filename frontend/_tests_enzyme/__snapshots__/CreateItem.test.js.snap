// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateItem/> renders and matches snapshot 1`] = `
<form
  className="FormStyles__StyledForm-txoayt-1 lkSerF"
  data-test="form"
  onSubmit={[Function]}
>
  <Spinner>
    <Spinner__StyledSpinner>
      <div
        className="Spinner__StyledSpinner-ma7w1f-0 FULCn"
      >
        <div
          id="backDrop"
        >
          <div
            id="loading"
          />
        </div>
      </div>
    </Spinner__StyledSpinner>
  </Spinner>
  <fieldset
    aria-busy={false}
    disabled={false}
  >
    <h2
      id="create_header"
    >
      Upload item
    </h2>
    <label
      htmlFor="file"
    >
      Image
      <input
        id="file"
        name="file"
        onChange={[Function]}
        placeholder="upload an image"
        type="file"
      />
    </label>
    <label
      htmlFor="title"
    >
      Title
      <input
        id="title"
        name="title"
        onChange={[Function]}
        placeholder="Title"
        required={true}
        type="text"
        value=""
      />
    </label>
    <label
      htmlFor="price"
      id="priceLabel"
    >
      Price
      <input
        id="price"
        name="price"
        onChange={[Function]}
        placeholder="Price"
        required={true}
        step=".01"
        type="number"
        value={0}
      />
    </label>
    <label
      htmlFor="description"
    >
      Description
      <textarea
        id="description"
        name="description"
        onChange={[Function]}
        placeholder="Enter a Description"
        required={true}
        value=""
      />
    </label>
    <button
      aria-disabled={false}
      disabled={false}
      type="submit"
    >
       Submit 
    </button>
  </fieldset>
</form>
`;
